lib/cron/fade-rate.ts

```diff
@@ -161,33 +161,52 @@ function createDynamoEntity() {
 
 const CREATE_VIEW_SQL = `
 CREATE OR REPLACE VIEW rfqOrders 
-AS
+AS (
+WITH latestOrders AS (
+  SELECT * FROM (
+    SELECT *, ROW_NUMBER() OVER (PARTITION BY filler ORDER BY createdat DESC) AS row_num FROM postedorders
+  )
+  WHERE row_num <= 20
+)
 SELECT
-    postedorders.chainid as chainId, postedorders.filler as rfqFiller, postedorders.quoteid, archivedorders.filler as actualFiller, postedorders.createdat as postTimestamp, archivedorders.txhash as txHash
+    latestOrders.chainid as chainId, latestOrders.filler as rfqFiller, latestOrders.quoteid, archivedorders.filler as actualFiller, latestOrders.createdat as postTimestamp, archivedorders.txhash as txHash,
+    CASE
+      WHEN latestOrders.inputstartamount = latestOrders.inputendamount THEN 'EXACT_INPUT'
+      ELSE 'EXACT_OUTPUT'
+    END as tradeType, 
+    CASE
+      WHEN latestOrders.inputstartamount = latestOrders.inputendamount THEN latestOrders.outputstartamount
+      ELSE latestOrders.inputstartamount
+    END as quotedAmount,
+    archivedorders.amountout as filledAmount
 FROM
-    postedorders LEFT OUTER JOIN archivedorders ON postedorders.quoteid = archivedorders.quoteid
+    latestOrders LEFT OUTER JOIN archivedorders ON latestOrders.quoteid = archivedorders.quoteid
 where
 rfqFiller IS NOT NULL
-AND postedorders.quoteId IS NOT NULL
+AND latestOrders.quoteId IS NOT NULL
 AND rfqFiller != '0x0000000000000000000000000000000000000000'
 AND chainId NOT IN (5,8001,420,421613) -- exclude mainnet goerli, polygon goerli, optimism goerli and arbitrum goerli testnets 
 AND
-    postTimestamp >= extract(epoch from (GETDATE() - INTERVAL '24 HOURS'));
+    postTimestamp >= extract(epoch from (GETDATE() - INTERVAL '168 HOURS')) -- 7 days rolling window
+);
 `;
 
 const FADE_RATE_SQL = `
 WITH ORDERS_CTE AS (
     SELECT 
         rfqFiller,
-        COUNT(*) AS TotalFills,
-        SUM(CASE WHEN rfqFiller != actualFiller THEN 1 ELSE 0 END) AS UnmatchedFills
+        COUNT(*) AS totalQuotes,
+        SUM(CASE WHEN rfqFiller != actualFiller OR (tradeType = 'EXACT_INPUT' AND quotedAmount > filledAmount) OR (tradeType = 'EXACT_OUTPUT' AND quotedAmount < filledAmount) THEN 1 ELSE 0 END) AS fadedQuotes
```

Comment:
I think `rfqFiller != actualFiller OR ` should be removed. You will still count the overrides in this case.